### ISSUE TITLE
`TextArea` should use `contenteditable` to allow for non text elements to be treated as text

### ISSUE THREAD

**Issue by databasedav** [MoonZoon/MoonZoon#156]
https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/contenteditable

**Comment by MartinKavik** [comment]
Could you elaborate a bit? I'm not really sure what problem do you want to resolve.

Regarding `contenteditable`: I was trying to use it months/years ago in a MoonZoon project and I gave up because of browser inconsistency and other problems if I remember correctly.

**Comment by databasedav** [comment]
the entire motivation is that `contenteditatble` allows non-text elements to be treated like text inline in a text box, so anything from an image to a custom data selector of some kind, for example, i could type "`/date`" or something in the `TextArea` and it could transform into a nice keyboard navigable date selector inline that has some special features like being able to output to different date formats on enter or restrict the dates that can be selected, etc.

**Comment by MartinKavik** [comment]
I don't remember I've seen the combination `TextArea` + `contenteditable` like you described and google doesn't want to show me any examples. Could you provide a link to a working example? (JS or a MZ example if you have one).

In most cases, people just trying to make standard HTML elements like `div` editable with `contenteditable` to create an in-place editor. But I would recommend to just replace the `div` with a standard `input` / `textarea` in this case.

**Comment by databasedav** [comment]
right, I was suggesting the latter, i.e. `TextArea` -> a `div` with `contenteditable`, in order to enable the benefits highlighted in my above comment
